src/main/java/com.company/bitmanipulation/XorSum.java

```diff
@@ -0,0 +1,43 @@
+package com.company.bitmanipulation;
+
+public class XorSum {
+
+    // 3, 5, 6, 8, 2
+    public int xorSumAllPairs(int[] arr) {
+        int bitLength = getBitLengthOfMaxLength(arr);
+        int answer = 0;
+        for (int i = 0; i < bitLength; i++) {
+            int ones = 0;
+            int zeros = 0;
+            for (int value : arr) {
+                if (((value >> i) & 1) == 1) {
+                    ones++;
+                } else {
+                    zeros++;
+                }
+            }
+
+            answer += (1 << i) * ones * zeros;
+        }
+        return answer << 1;
+    }
+
+    public int getBitLengthOfMaxLength(int[] arr) {
```

Comment:
Not mandatory: You can move this method into any of the util or helper class as I see this same code multiple places.

Reply:
sure will move to it helper class